workflow-gen/Program.cs

```diff
@@ -0,0 +1,123 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace WorkflowGen
+{
+    using Dapr.Client;
+    using Dapr.Workflow;
+    using Dapr.Tests.Common.Models;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.Hosting;
+    using Prometheus;
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using WorkflowGen.Activities;
+    using WorkflowGen.Models;
+    using WorkflowGen.Workflows;
+    /// <summary>
+    /// WorkflowGenerator - runs worflows and executes them using Dapr.
+    /// The main functionality is in StartWorkflowGeneratorAsync().
+    /// </summary>
+    public class Program
+    {
+        private static readonly Gauge ExecutionCallTime = Metrics.CreateGauge("lh_workflow_generator_execution_call_time", "The time it takes for the workflow call to return");
+
+        private static readonly Counter ExecutionFailureCount = Metrics.CreateCounter("lh_workflow_generator_execution_failure_count", "Publich calls that throw");
+
+        const string DaprWorkflowComponent = "dapr";
+
+
+        /// <summary>
+        /// Main for WorkflowGen
+        /// </summary>
+        /// <param name="args">Arguments.</param>
+        public static void Main(string[] args)
+        {
+            int delayInMilliseconds = 10000;
+
+            var server = new MetricServer(port: 9988);
+            server.Start();
+
+            var builder = Host.CreateDefaultBuilder(args).ConfigureServices(services =>
+            {
+                services.AddDaprWorkflow(options =>
+                {
+                    options.RegisterWorkflow<OrderProcessingWorkflow>();
+                    options.RegisterActivity<NotifyActivity>();
+                    options.RegisterActivity<ReserveInventoryActivity>();
+                    options.RegisterActivity<ProcessPaymentActivity>();
+                    options.RegisterActivity<UpdateInventoryActivity>();
+                });
+            });
+
+            Task.Run(() => StartWorkflowGeneratorAsync(delayInMilliseconds));
+
+            using var host = builder.Build();
+            host.Run();
+
+
+        }
+
+        static internal async void StartWorkflowGeneratorAsync(int delayInMilliseconds)
```

Comment:
For this kind of method, I'd recommend using a `Timer` like we do in the `pubsub workflow`. If you wanted to get fancier, you could even move the generation code to a service and scope everything out with DI like I do in this PR: https://github.com/dapr/test-infra/pull/173/files#diff-bfa58cd42c1956bd37e1c2a1d5529cae1ec18ee93956472f80dfdf66dd071d6a.

At a minimum though, we should switch to the `Timer` instead of just doing the `while(true)` with a delay.